airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence2.java

```diff
@@ -0,0 +1,1911 @@
+    } else if (configType == ConfigSchema.SOURCE_CONNECTION) {
+      writeSourceConnection(configs.values().stream().map(c -> (SourceConnection) c).collect(Collectors.toList()));
+    } else if (configType == ConfigSchema.DESTINATION_CONNECTION) {
+      writeDestinationConnection(configs.values().stream().map(c -> (DestinationConnection) c).collect(Collectors.toList()));
+    } else if (configType == ConfigSchema.SOURCE_OAUTH_PARAM) {
+      writeSourceOauthParameter(configs.values().stream().map(c -> (SourceOAuthParameter) c).collect(Collectors.toList()));
+    } else if (configType == ConfigSchema.DESTINATION_OAUTH_PARAM) {
+      writeDestinationOauthParameter(configs.values().stream().map(c -> (DestinationOAuthParameter) c).collect(Collectors.toList()));
+    } else if (configType == ConfigSchema.STANDARD_SYNC_OPERATION) {
+      writeStandardSyncOperation(configs.values().stream().map(c -> (StandardSyncOperation) c).collect(Collectors.toList()));
+    } else if (configType == ConfigSchema.STANDARD_SYNC) {
+      writeStandardSync(configs.values().stream().map(c -> (StandardSync) c).collect(Collectors.toList()));
+    } else if (configType == ConfigSchema.STANDARD_SYNC_STATE) {
+      writeStandardSyncState(configs.values().stream().map(c -> (StandardSyncState) c).collect(Collectors.toList()));
+    } else {
+      throw new IllegalArgumentException("Unknown Config Type " + configType);
+    }
+  }
+
+  @Override
+  public void deleteConfig(AirbyteConfig configType, String configId) throws ConfigNotFoundException, IOException {
+    if (!useNewPersistence()) {
+      oldConfigPersistence.deleteConfig(configType, configId);
+      return;
+    }
+    if (configType == ConfigSchema.STANDARD_WORKSPACE) {
+      deleteStandardWorkspace(configId);
+    } else if (configType == ConfigSchema.STANDARD_SOURCE_DEFINITION) {
+      deleteDefinition(configId);
+    } else if (configType == ConfigSchema.STANDARD_DESTINATION_DEFINITION) {
+      deleteDefinition(configId);
+    } else if (configType == ConfigSchema.SOURCE_CONNECTION) {
+      deleteConnection(configId);
+    } else if (configType == ConfigSchema.DESTINATION_CONNECTION) {
+      deleteConnection(configId);
+    } else if (configType == ConfigSchema.SOURCE_OAUTH_PARAM) {
+      deleteOauthParams(configId);
+    } else if (configType == ConfigSchema.DESTINATION_OAUTH_PARAM) {
+      deleteOauthParams(configId);
+    } else if (configType == ConfigSchema.STANDARD_SYNC_OPERATION) {
+      deleteStandardSyncOperation(configId);
+    } else if (configType == ConfigSchema.STANDARD_SYNC) {
+      deleteStandardSync(configId);
+    } else if (configType == ConfigSchema.STANDARD_SYNC_STATE) {
+      deleteStandardSyncState(configId);
+    } else {
+      throw new IllegalArgumentException("Unknown Config Type " + configType);
+    }
+  }
+
+  private void deleteStandardWorkspace(final String configId) throws IOException {
```

Comment:
is it possible to dedupe all of the deletions. fundamentally they are all doing `DELETE FROM <table name> WHERE ID=?`. I know there are one or 2 exceptions but with this approach i think you can remove a lot of lines of code.

Reply:
+1